src/Typedefs.idr

```diff
@@ -149,6 +162,9 @@ mutual
   Ty     tvars (TVar v)    = Vect.index v tvars
   Ty     tvars (TMu m)     = Mu tvars (args m)
   Ty     tvars (TApp f xs) = assert_total $ Ty tvars $ def f `ap` xs -- TODO: could be done properly
+  -- either use a proof that no TRefs exist or a proof that every 
+  -- TRef has a mapping to an Idris type
+  Ty     tvars (TRef n)    = Void
```

Comment:
This is where scoped references would be very helpful to do the right thing.

Reply:
That's a very good point, I don't feel like handling errors at this stage of the pipeline is wise.

We should have a vector that verifies that all those references do actually exist and access them without worrying about throwing errors. The vector should be built in an earlier stage which could throw errors (for example if a specialized type isn't available on the platform we're targetting)